cpp/src/groupby/hash/groupby.cu

```diff
@@ -135,6 +137,8 @@ class hash_compound_agg_finalizer final : public cudf::detail::aggregation_final
       stream(stream),
       mr(mr)
   {
+    result_type = cudf::is_dictionary(col.type()) ? cudf::dictionary_column_view(col).keys().type()
```

Comment:
For groupby, I wonder if it really makes sense for the output to also be dictionary encoded. Typically the result is relatively small and the values unique.  Dictionary encoding such a column would almost certainly just be pure overhead without any benefit.

Reply:
Oh, nevermind, it isn't dictionary encoding it. Carry on.